Bigtable/src/Table.php

```diff
@@ -0,0 +1,351 @@
+     * @see https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#table Table
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function reload(array $options = [])
+    {
+        throw new \BadMethodCallException('This method is not implemented yet');
+    }
+
+    /**
+     * Return the table state.
+     *
+     * When tables are created or updated, they may take some time before
+     * they are ready for use. This method allows for checking whether a
+     * table is ready.
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function state(array $options = [])
+    {
+        throw new \BadMethodCallException('This method is not implemented yet');
+    }
+
+    /**
+     * Create a table.
+     *
+     * Example:
+     * ```
+     * $tableInfo = $table->create();
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.CreateTableRequest CreateTableRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @todo add options for ColumnFamily
+     * @param array $options [optional] Configuration options.
+     * @return array Table information
+     */
+    public function create(array $options = [])
+    {
+        $instanceName = TableAdminClient::instanceName($this->projectId, $this->instanceId);
+        $this->info = $this->connection->createTable([
+            'parent' => $instanceName,
+            'tableId' => $this->id
+        ] + $options);
+
+        return $this->info;
+    }
+
+    /**
+     * Delete the current table.
+     *
+     * Example:
+     * ```
+     * $table->delete();
+     * ```
+     *
+     * @param array $options [optional] Configuration options.
+     *
+     * @return void
+     */
+    public function delete(array $options = [])
+    {
+        return $this->connection->deleteTable([
+            'name' => $this->name
+        ] + $options);
+    }
+
+    /**
+     * Add column families to the current table.
+     *
+     * Example:
+     * ```
+     * $tableInfo = $table->addColumnFamilies([
+     *     ['id' => 'cf1', 'gcRule'=>[]],
+     *     ['id' => 'cf2', 'gcRule'=>[]],
+     *     ['id' => 'cf3', 'gcRule'=>[]]
+     * ]);
+     * ```
+     * @todo add GCRule options
+     * @param array[] $columnFamilies List of ColumnFamilies, each entry in the list
+     *        is array with key `id` for id of ColumnFamily
+     * @param array $options [optional] Configuration options.
+     * @return array Table information
+     */
+    public function addColumnFamilies(array $columnFamilies, array $options = [])
+    {
+        $modifications = [];
+        foreach ($columnFamilies as $columnFamily) {
+            $modifications[] = ['id' => $columnFamily['id'], 'create' => $columnFamily['gcRule']];
```

Comment:
I'm not sure whether our previous discussion about validating the `$modifications` array was addressed yet. I was thinking it could be handled here.

```php
        foreach ($columnFamilies as $columnFamily) {
            if (!isset($columnFamily['id']) {
                throw new \InvalidArgumentException('Each column family must specify an ID.');
            }

            $rule = isset($columnFamily['gcRule']) && is_array($columnFamily['gcRule'])
                ? $columnFamily['gcRule']
                : []
            $modifications[] = ['id' => $columnFamily['id'], 'create' => $rule];
        }
```

Assuming `gcRule` can be empty, even once the client is complete, we could make it optional and simply default to an empty array. If that's not the case, an additional check for the presence of the `gcRule` key can replace the ternary statement.